### DISCUSSION TITLE
v0.3.3

### DISCUSSION THREAD

**Discussion by smaye81** [connectrpc/connect-es#363] [Announcements]
This release includes the following:

## Enhancements
* Upgrade to protobuf-es v0.4.0 by @smaye81 #361

<hr /><em>This discussion was created from the release <a href='https://github.com/bufbuild/connect-web/releases/tag/v0.3.3'>v0.3.3</a>.</em>

**Comment by scvnc** [comment]
@smaye81 

I'm using connect-web in a javascript [monorepo](https://lerna.js.org/) and this versioning may have blown things up on my end.

I have a package that is dependent on `0.3.1` that I wasn't ready to upgrade yet and a new package that is using `0.3.3` (with types generated by `@bufbuild/protoc-gen-es@0.4.0`)

Yarn/lerna correctly hoisted `@bufbuild/connect-web` to the root because `0.3.1` and `0.3.3` should be compatible.  This is because the semver indicates it's only the patch version and should be equivalent.  

However I am seeing some typescript mismatch errors when trying to do the `createPromiseClient` as `@bufbuild/connect-web@0.3.1` is designed to work with the types emitted from `@bufbuild/protoc-gen-es@0.2.1` while `@bufbuild/connect-web@0.3.3` is designed to work with the types emitted from `@bufbuild/protoc-gen-es@0.4.0`.

I suspect that this package should have been released as `0.4.0` to signal to the package resolvers that it is not merely a patched version of `0.3.1`

Thanks for all your help!!! :)

**Comment by smaye81** [comment]
Hi @scvnc.  It makes sense what you're saying.  Just to confirm though, what sort of errors are you seeing?  Can you paste them here?

**Reply by scvnc** [reply]
It's surrounding the code `createPromiseClient(GeneratedService, transport)`  complaining that GeneratedService is not compatible with the type signature for the first argument of `createPromiseClient`

Where `GeneratedServiceDef` was built off a proto via `@bufbuild/protoc-gen-es@0.2.1`

I would show the whole typescript error, but I would have to spend a lot of time readacting properties of my proto which I'm not authorized to share. 

![image](https://user-images.githubusercontent.com/885196/206232199-86f0f19b-bb18-423b-8d67-0c66883046fa.png)

![image](https://user-images.githubusercontent.com/885196/206231564-a49d8920-b1d7-4fbf-ad1b-bdf4f5bf4b80.png)

![image](https://user-images.githubusercontent.com/885196/206229787-c70fd357-a0d5-4076-ab90-0004b60dca54.png)

It's resolved for me now as I've upgraded the other packages to use a generated service based off of `@bufbuild/protoc-gen-es@0.4.0` such that it's all compatible when yarn/lerna hoists/dedups to `@bufbuild/connect-web@0.3.3`

My apologies for the screen shots as opposed to text.  I do not have the error readily available anymore since I have upgraded the packages across my system.

**Comment by smaye81** [comment]
Hey @scvnc sorry for the delayed response.  I see what you mean.  I'm not entirely sure how this happened, but I suspect you're correct in your assumption.  Working with npm in monorepos is pretty difficult.  I'm glad you got your issue resolved, but we will investigate this further for the next release to hopefully resolve the issue going forward.
